shared/crossCompilation/rust.md

```diff
@@ -0,0 +1,50 @@
+#### Setting up {{ $language.name }} cross-compilation
+
+Cross-compilation is needed when using `resin sync` and {{ $language.name }} as it is a compiled language. Compiled languages are programming languages in which source code must be compiled into an executable form before it can be run.
+
+Luckily {{ $language.name }} cross-compilation is fairly easy to get setup using [rust-cross][rust-cross] and even easier to use.
+
+To get started we first install {{ $language.name }} on our development machine. [rustup.rs][rustup.rs] makes this a lot easier and is highly recommended.
+```
+$ curl https://sh.rustup.rs -sSf | sh
+```
+Next up is the C cross-compilation toolchain.
+```
+$ sudo apt-get install -qq gcc-arm-linux-gnueabihf
+```
+Now we need to install a standard set of crates that have been cross-compiled for the {{ $device.name }}. A crate is synonymous with a ‘library’ or ‘package’ in other languages.
+```
+$ rustup target add {{ $device.triple }}
+```
+Finally {{ $language.name }} must be configured to enable cross-compilation for the {{ $device.name }}.
+```
+$ mkdir -p ~/.cargo
+$ cat >>~/.cargo/config <<EOF
+> [target.{{ $device.triple }}]
+> linker = "{{ $device.triple }}"
+> EOF
+```
+
+Now that cross-compilation is set up it would be wise to test everything works before moving on to configuring `resin sync`. The commands below will create a new {{ $language.name }} project and cross-compile it for your {{ $device.name }}.
+```
+$ cargo new --bin hello
+$ cd hello
+$ cargo build --target={{ $device.triple }}
+   Compiling hello v0.1.0 (file:///home/ubuntu/hello)
+$ file target/{{ $device.triple }}/debug/hello
+hello: ELF 32-bit LSB  shared object, ARM, EABI5 version 1 (SYSV), dynamically linked (uses shared libs), for GNU/Linux 2.6.32, BuildID[sha1]=67b58f42db4842dafb8a15f8d47de87ca12cc7de, not stripped
+```
+Copy the output file `hello` from the directory `target/{{ $device.triple }}/debug/` to your device and ensure it runs correctly.
```

Comment:
how do we recommend they copy it here? at this point they probably don't know about resin ssh and most likely don't have openSSH setup in their project so copying may be a difficult task